### ISSUE TITLE
Allow SingleFile to operate on restricted pages (in Firefox).

### ISSUE THREAD

**Issue by RokeJulianLockhart** [gildas-lormeau/SingleFile#1494]
#### **Is your feature request related to a problem? Please describe.**

Although https://github.com/gildas-lormeau/SingleFile/blob/14da84df6dd3191b31f45c6ecb1dc5c721eea090/faq.md#why-cant-i-save-some-pages-like-httpsaddonsmozillaorgaddonsingle-file states:

> For security purposes, browsers block web extensions on certain domains. This prevents a malicious extension to remove or change bad reviews, for example.

This is bypassable now. As https://superuser.com/questions/1354603/firefox-extensions-restricted-from-running-on-some-sites#comment2926608_1354612 explains, https://support.mozilla.org/en-US/kb/quarantined-domains?as=u&utm_source=inproduct#:~:text=how%20can%20i%20re-enable%20the%20add-ons%20that%20are%20not%20allowed%20on%20sites%20restricted%20by%20mozilla%3F demonstrates that this restriction can be disabled by the user via `about:addons` for add-ons which opt-in.

#### **Describe the solution you'd like**

Provide this preference.

**Comment by gildas-lormeau** [comment]
I agree this is possible in Firefox. In fact this problem also exists in Chrome and (probably) Safari and I guess I didn't want to indicate every procedure for every browser because it's a problem that's supposed to be rare, and for advanced users. It's also quite easy to get around by using another browser.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/gildas-lormeau/SingleFile/issues/1494#issuecomment-2215255420

It's easy to get around for the content of `extensions.webextensions.restrictedDomains`:

> ```
> accounts-static.cdn.mozilla.net,accounts.firefox.com,addons.cdn.mozilla.net,addons.mozilla.org,api.accounts.firefox.com,content.cdn.mozilla.net,discovery.addons.mozilla.org,install.mozilla.org,oauth.accounts.firefox.com,profile.accounts.firefox.com,support.mozilla.org,sync.services.mozilla.com
> ```

...but not `about:` schema-prefixed URIs.

**Comment by gildas-lormeau** [comment]
I'm moving this issue in the Discussions tab because it's not possible from a technical point of view to implement this option in the extension itself.
